DataDog/libdatadog
# [PROF-8917] Fix case on crashtracker ffi APIs + add Ruby helper + add example

# What does this PR do?

This PR contains a bag of changes to the crashtracker that I worked on as part of integrating it into Ruby. See individual commits for details.

# Motivation

N/A

# Additional Notes

N/A

# How to test the change?

Changes include test coverage; furthermore, the new example can be used to see the crash tracker in action and the downstream Ruby PR (https://github.com/DataDog/dd-trace-rb/pull/3384) also can be used to see this in action.

## For Reviewers
